src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -64,19 +65,34 @@ public ObjectResult(object value)
         {
             var incomingAcceptHeader = HeaderParsingHelpers.GetAcceptHeaders(
                                                 formatterContext.ActionContext.HttpContext.Request.Accept);
-            var sortedAcceptHeaders = SortMediaTypeWithQualityHeaderValues(incomingAcceptHeader)
+            var sortedAcceptHeaderMediaTypes = SortMediaTypeWithQualityHeaderValues(incomingAcceptHeader)
                                         .Where(header => header.Quality != HttpHeaderUtilitites.NoMatch)
                                         .ToArray();
 
+            var options = formatterContext.ActionContext.HttpContext
+                                                        .RequestServices
+                                                        .GetRequiredService<IOptions<MvcOptions>>()
+                                                        .Options;
+
             IOutputFormatter selectedFormatter = null;
 
+            bool ignoreAcceptHeader = false;
+            if (options.IgnoreBrowserAcceptHeader
+                    && HasWildCardMediaAndSubMediaType(sortedAcceptHeaderMediaTypes))
+            {
+                ignoreAcceptHeader = true;
```

Comment:
It's typically better to use positive logic, so perhaps rename to RespectBrowserAcceptHeader and respectAcceptHeader respectively